for_dict_challenges_bonus.py

```diff
@@ -65,6 +65,92 @@ def generate_chat_history():
         })
     return messages
 
+# 1. Вывести айди пользователя, который написал больше всех сообщений.
+def user_wrote_most_messages(history):
+    
+    user_id_message = {}  # создаётся словарь с парой - 'user_id': кол-во сообщений
+    for user_id in history:
+        user_id_message[user_id['sent_by']] = user_id_message.get(user_id['sent_by'], 0) + 1
+
+    user_id_max = max(user_id_message.items(), key=lambda items: items[1])  # определяю максимальное количество сообщений
+    
+    users_id_max = [] 
+    for key, value in user_id_message.items():  # для определения всех айди пользователей, которые написали больше всех сообщений
+        if value >= user_id_max[1]:
+            users_id_max.append(key)
+
+    return f'ID пользователя(ей), который(е) написал(и) больше всех сообщений: {users_id_max}'
+
+# 2. Вывести айди пользователя, на сообщения которого больше всего отвечали.
+def user_messages_most_answered(history):
+
+    message_db = {}
+    for message in history:  # создаём словарь ключ (id message): значение (весь message)
+        message_db[message['id']] = message
+    users_counter = {}
+    for message in message_db.values():  # перебираем значения, созданного выше словаря message_db
+        if message['reply_for'] is not None:  # если ответ на сообщение пользователя не None
+            parent_message = message  # в переменную parent_message передаётся весь message
+            user_id = parent_message['sent_by']  # в переменную user_id передаётся ID пользователя
+            users_counter[user_id] = users_counter.get(user_id, 0) + 1  # создаётся словарь ключ (ID пользователя): значение (количество сообщений, направленных ему)
+    
+    user_id_max = max(users_counter.items(), key=lambda items: items[1])  # определяется кортеж (ID пользователя, максимальное количество сообщений)
+    
+    users_id_max = [] 
+    for key, value in users_counter.items():  # для определения всех айди пользователей (в случае повторений максимального количества сообщений), на сообщения которого больше всего отвечали
+        if value == user_id_max[1]:
+            users_id_max.append(key)
+
+    return f'ID пользователя(ей), на сообщения которого(ых) больше всего отвечали: {users_id_max}'
+
+# 3. Вывести айди пользователей, сообщения которых видело больше всего уникальных пользователей.
+def id_users_posts_seen_most_uniq_users(history):
+    messages_db = {}
+    for message in history:
+        messages_db[message['id']] = message
+    
+    id_user_uniq_users = {}  # создаём словарь ключ (id_messege): значение (количество уникальных пользователей)
+    for mess in messages_db.values():
+        id_user_uniq_users[mess['id']] = mess['seen_by'] = len(mess['seen_by']) # замена значения по ключу seen_by в словаре messages_db
```

Comment:
```suggestion
        id_user_uniq_users[mess['id']] =  len(mess['seen_by']) # замена значения по ключу seen_by в словаре messages_db
```
Затирать список  seen_by - осуждаю, так как не вижу совершенно никакой причины это делать. Можно потом долго выискивать невнятные ошибки в коде, потому что кто-то другой будет ожидать что в seen_by лежит список, а не число.